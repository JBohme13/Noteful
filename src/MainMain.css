#notes-main {
    height: 100px;
    text-align: center;
    margin-top: 2.5px;
    margin-bottom: 2.5px;
    border: 1px grey solid;
    display: flex;
    flex-direction: column;
    box-sizing:border-box;
    justify-content: space-evenly;
}

#note-modified {
    color: darkslategrey;
}

#note-link {
  text-decoration: none;
  color: darkslategrey;
  font-size: 24px;
}

#delete-button {
    width: 72px;
    height: 22px;
    margin: 5px;
    border: 1px solid grey;
    border-radius: 2px;
    box-shadow: 1px 1px;
    font-size: 14px;
    color: darkslategrey;
}

#add-note {
    text-decoration: none;
    color: darkslategrey;
}

#add-note-container:active {
    box-shadow: -1px -1px;
}

#add-note-container {
    border: 1px solid grey;
    border-radius: 3px;
    box-shadow: 1px 1px;
    margin: 5px;
    height: 22px;
    font-size: 14px;
}