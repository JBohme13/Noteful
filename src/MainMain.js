import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import NotefulContext from './NotefulContext'
import PropTypes from 'prop-types'
import './MainMain.css'

export default class MainMain extends Component{
    static contextType = NotefulContext;
    render() {
        const value = this.context;
        const renderNotes = value.notes.map((note, i) => {
          return(
              <section className='notes-main' key={i}>
                <Link 
                  to={`/notes/${note.id}`} 
                  id='note-link'
                  onClick={() => {
                    value.setNoteId(note.id);
                    value.setFolderId(note.folderId);
                  }}
                >
                  {note.name}
                </Link>
                <span 
                  className='note-modified'>{new Date(note.modified).toLocaleString()}
                </span>
                  <button 
                    id='delete-button'
                    onClick={e => value.deleteNote(note.id)}
                  >
                    Delete
                  </button>
              </section>
          )}
        )
        return(
            <section className='main-container'>
                { renderNotes }
              <button 
                id='add-note-container'
                onClick ={() => value.history.push('/add-note')}>
                  Add Note
              </button>
            </section>
        )
    }
};

MainMain.propTypes = {
  notes: PropTypes.array,
  setNoteId: PropTypes.func,
  setFolderid: PropTypes.func,
  deleteNotes: PropTypes.func,
};