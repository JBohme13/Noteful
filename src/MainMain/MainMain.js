import React, { Component } from 'react'
import NotefulContext from './NotefulContext'
import NoteError from './NoteError'
import DeleteError from './DeleteError'
import './MainMain.css'

export default class MainMain extends Component{
    static contextType = NotefulContext;
    render() {
        const value = this.context;
        const notes = value.notes.map((note, i) => 
            <NoteError>
              <section id='notes-main' key={i}>
                <a 
                  href={`/notes/${note.id}`} 
                  id='note-link'
                  onClick={() => value.setNoteId(note.id)}
                >
                  {note.name}
                </a>
                <span id='note-modified'>{new Date(note.modified).toLocaleString()}</span>
                <DeleteError>
                  <button 
                    id='delete-button'
                    onClick={e => value.deleteNote(note.id)}
                  >
                    Delete
                  </button>
                </DeleteError>
              </section>
            </NoteError>
        )
        return(
            <div className='main-container'>
                { notes }
              <button 
                id='add-note-container'
                onClick ={() => value.history.push('/add-note')}>
                <a 
                  href={'/add-note'}
                  id='add-note'
                >
                  Add Note
                </a>
              </button>
            </div>
        )
    }
}