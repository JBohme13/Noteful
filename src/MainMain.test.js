import React from 'react';
import ReactDOM from 'react-dom';
import MainMain from './MainMain';
import renderer from 'react-test-renderer'

describe('MainMain component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MainMain />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<MainMain />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
});