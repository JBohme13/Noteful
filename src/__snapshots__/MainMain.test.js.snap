// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMain component renders the UI as expected 1`] = `
<div
  className="main-container"
>
  <button
    id="add-note-container"
    onClick={[Function]}
  >
    <a
      href="/add-note"
      id="add-note"
    >
      Add Note
    </a>
  </button>
</div>
`;
